src/samples.jl

```diff
@@ -0,0 +1,121 @@
+function load_distributions_samples()
+return quote
+
+using .Distributions: Uniform, Normal
+
+"""
+    _sample_unit_nsphere_muller!(D::Vector{Vector{N}}, n, p;
+                                 rng::AbstractRNG=GLOBAL_RNG,
+                                 seed::Union{Int, Nothing}=nothing) where {N}
+
+Return samples from a uniform distribution on an ``n``-dimensional unit
+sphere using Muller's method.
+
+### Input
+
+- `D`    -- output, vector of vectors
+- `n`    -- dimension of the sphere
+- `p`    -- number of random samples
+- `rng`  -- (optional, default: `GLOBAL_RNG`) random number generator
+- `seed` -- (optional, default: `nothing`) seed for reseeding
+
+### Output
+
+A vector of `nsamples` vectors.
+
+### Algorithm
+
+This function implements Muller's method of normalised Gaussians [1] to uniformly
+sample over the ``n``-dimensional sphere ``S^n`` (which is the bounding surface
+of the ``n``-dimensional unit ball).
+
+Given ``n`` canonical Gaussian random variables ``Z₁, Z₂, …, Z_n``, the
+distribution of the vectors
+
+```math
+\\begin \\dfrac{1}{α} \\left( z₁, z₂, …, z_n \\right)^\\transpose,
+```
+where ``α := \\sqrt\\{z₁² + z₂² + … + z_n²\\}``, is uniform over ``S^n``.
+
+[1] Muller, Mervin E. *A note on a method for generating points uniformly on
+    n-dimensional spheres.* Communications of the ACM 2.4 (1959): 19-20.
+"""
+function _sample_unit_nsphere_muller!(D::Vector{Vector{N}}, n, p;
+                                      rng::AbstractRNG=GLOBAL_RNG,
+                                      seed::Union{Int, Nothing}=nothing) where {N}
+    rng = reseed(rng, seed)
```

Comment:
The functions in this file, such as `_sample_unit_nsphere_muller!`, are already inside the `load_distributions_samples()` block, so we don't need `require(:Distributions)` because they are not visible without it (same for functions in other `load_...` blocks).